en/advanced_config/tuning_the_ecl_ekf.md

```diff
@@ -128,16 +128,22 @@ This minimum data set is required for all EKF modes of operation. Other sensor d
 ### Magnetometer
 
 Three axis body fixed magnetometer data (or external vision system pose data) at a minimum rate of 5Hz is required.
+
 Magnetometer data can be used in two ways:
 
-* Magnetometer measurements are converted to a yaw angle using the tilt estimate and magnetic declination.
-  This yaw angle is then used as an observation by the EKF.
-  This method is less accurate and does not allow for learning of body frame field offsets, however it is more robust to magnetic anomalies and large start-up gyro biases.
-  It is the default method used during start-up and on ground.
-* The XYZ magnetometer readings are used as separate observations.
-  This method is more accurate and allows body frame offsets to be learned, but assumes the earth magnetic field environment only changes slowly and performs less well when there are significant external magnetic anomalies.
+- Magnetometer measurements are converted to a yaw angle using the tilt estimate and magnetic declination.
+  The yaw angle is then used as an observation by the EKF.
+  - This method is less accurate and does not allow for learning of body frame field offsets, however it is more robust to magnetic anomalies and large start-up gyro biases.
+  - It is the default method used during start-up and on ground.
+- The XYZ magnetometer readings are used as separate observations.
+  - This method is more accurate when the vehicle is moving, as it allows body frame offsets to be learned (when the vehicle is moving the heading or the magnetometer biases are observable).
```

Comment:
This is correct?

Reply:
Not exactly. The method is more accurate (no need to move to be more accurate) but it requires to have the mag bias correctly estimated. And to estimate the biases, you need to move. Once they are estimated and if they don't change anymore, the 3D fusion is still more accurate even while hovering. However, since we cannot guarantee that the bias are not changing and that the bias are no longer observable while not moving, it is safer to switch back to heading fusion when the motion stops.